metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -482,7 +491,18 @@ final class FileDecoderProvider(
   ): Future[DecoderResponse] = {
     val cfrDependency = Dependency.of("org.benf", "cfr", "0.151")
     val cfrMain = "org.benf.cfr.reader.Main"
-    val args = List("--analyseas", "CLASS", s"""${path.toNIO.toString}""")
+    val (parent, className) =
+      if (path.isJarFileSystem)
+        (workspace, path.toNIO.toString.stripPrefix("\\").stripPrefix("/"))
+      else (path.parent, path.toNIO.toString)
+    val extraClassPath = path.jarPath
+      .map(jarPath => List("--extraclasspath", jarPath.toString))
+      .getOrElse(List.empty)
+    val args = extraClassPath ::: List(
+      "--analyseas",
```

Comment:
Typo?

Reply:
No - although it does look it.  The options is meant to mean ANALYSE_AS.  Either `jar` or `class`